### PR TITLE
Improve plugins index

### LINKED ISSUE #453: Adding (third-party) plugins to the index
For the beginning there will be only "official" plugins on the index / list under https://woodpecker-ci.org/plugins. The question that came up in #451 is how third-party plugins should be added and how they need to be managed to provide some kind of liability.

### LINKED ISSUE #859: List of community plugins
we have #315 for our own plugins ...

but it would be nice to just have some simple list to link to plugins like e.g.:
 - https://codeberg.org/qwerty287/woodpecker-intellij-formatter
 - https://codeberg.org/qwerty287/woodpecker-regex-check
 - ...

### LINKED ISSUE #1034: Replace plugin auto-detection with some kind of config file
We currently use the Github api to find all public repos of the woodpecker org by using `org:woodpecker-ci topic:woodpecker-plugin fork:true` as search term. This restricts us to only list plugins from one forge (Github) atm and also makes it difficult to allow plugins which are not part of the `woodpecker-ci` org (see #859).

Proposal:
To be more flexible I would suggest to replace this functionality and use a simple yaml config file instead. To get the same information we currently show following config file format could be used:

```yml
plugins:
  - repoUrl: https://github.com/woodpecker-ci/plugin-surge-preview
    docsUrl: https://raw.githubusercontent.com/woodpecker-ci/plugin-surge-preview/main/docs.md
    official: true # show a blue checkmark on the plugin list that this plugin can be trusted (add hint to all others to use at on own risk)
```

The docs url would link to a txt file with a format like this (already used):

[Source](https://raw.githubusercontent.com/woodpecker-ci/plugin-surge-preview/main/docs.md)

```md
---
name: Surge preview plugin
authors: Woodpecker Authors
icon: https://woodpecker-ci.org/img/logo.svg
description: Plugin to create static pages deployments as preview environments on pull-requests.
tags: [publish, cdn, preview]
image: woodpeckerci/plugin-surge-preview
---

The surge-preview plugin [...]

## Usage

[...]
```

### PR DESCRIPTION
**Pull request by anbraten**
image

image

closes #859 
closes #453
closes #1034

### CODE REVIEW

File: docs/docs/92-awesome.md [hunk 1]
```diff
@@ -25,9 +25,15 @@ If you have some missing resources, please feel free to [open a pull-request](ht
 
 ## Templates
 
-## Blogs, guides, videos
+## Posts & tutorials
 
-- [Tutorial about setup of Woodpecker-CI and Gitea](https://containers.fan/posts/setup-gitea-with-woodpecker-ci/)
+- [Setup Gitea with Woodpecker CI](https://containers.fan/posts/setup-gitea-with-woodpecker-ci/)
```
Comment (6543) [inline docs/docs/92-awesome.md:30]: oh nice where did you got that list from :)

Comment (anbraten) [reply docs/docs/92-awesome.md:30]: google :wink:

**Comment by anbraten** [comment]
That's what the `verified` label is for.

**Comment by anbraten** [comment]
Shouldn't be to hard to add a stamp icon. Give me some minutes :wink:

**Comment by anbraten** [comment]
Escalated a bit, but the new "design" got a tiny search and a few other improvements.

File: docs/plugins/woodpecker-plugins/src/theme/WoodpeckerPluginList.tsx [hunk 2]
```diff
@@ -34,47 +38,57 @@ export function WoodpeckerPluginList({ plugins }: { plugins: WoodpeckerPlugin[]
...
-          <div className="container">
-            <div className="row">
-              <NewPluginPanel />
-              {plugins.map((plugin, idx) => (
-                <PluginPanel key={idx} plugin={plugin} />
+          <div style={{ display: 'flex', flexFlow: 'column', alignItems: 'center' }}>
+            <h1>Woodpecker CI plugins</h1>
+            <p>This list contains plugins which you can use to easily execute usual pipeline tasks.</p>
+            <a href={applyForIndexUrl} target="_blank" rel="noopener noreferrer" className="button button--primary">
+              🙏 Please add your plugin
```
[the remark this answers is not in the document]
Comment (anbraten) [reply docs/plugins/woodpecker-plugins/src/theme/WoodpeckerPluginList.tsx:76]: You got some inspiration 😉, but we can change it

image

File: docs/plugins/woodpecker-plugins/package.json [hunk 3]
```diff
@@ -5,24 +5,27 @@
   "main": "dist/index.js",
   "typings": "dist/index.d.ts",
   "scripts": {
-    "start": "concurrently 'tsc -w' 'tsc -w -p tsconfig.jsx.json'",
-    "build": "tsc && tsc -p tsconfig.jsx.json"
+    "start": "pnpm run style && concurrently 'tsc -w' 'tsc -w -p tsconfig.jsx.json'",
+    "build": "pnpm run style && tsc && tsc -p tsconfig.jsx.json",
```
[the remark this answers is not in the document]
Comment (anbraten) [reply docs/plugins/woodpecker-plugins/package.json:9]: Has to be yarn. I am using `pnpm` for most other projects, mixing it up quite often 🤦🏾

[the remark this answers is not in the document]
Comment (anbraten) [reply docs/plugins/woodpecker-plugins/package.json:9]:
```suggestion
    "start": "yarn run style && concurrently 'tsc -w' 'tsc -w -p tsconfig.jsx.json'",
    "build": "yarn run style && tsc && tsc -p tsconfig.jsx.json",
```

File: docs/plugins/woodpecker-plugins/src/theme/WoodpeckerPluginList.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (anbraten) [reply docs/plugins/woodpecker-plugins/src/theme/WoodpeckerPluginList.tsx:76]:
```suggestion
              🎉 Add your own plugin
```

**Comment by anbraten** [comment]
Deployment: https://woodpecker-ci-woodpecker-pr-1200.surge.sh/plugins

---
Why is our bot not commenting here 🤔

**Pull request merged**